### DISCUSSION TITLE
Capturing a succession of statements between two comments.

### DISCUSSION THREAD

**Discussion by Klafyvel** [tree-sitter/tree-sitter#3638] [Q&A] [answered]
Hi! I am new to tree-sitter queries, and I am trying to capture the code placed between two comments. For example, with the following file:
```julia
# A comment that should be markdown.
# #retest

println("Second cell!")

function foo(arguments)
end

# Another comment that is markdown. [test](https://klafyvel.me)
```

I would like to capture the following code under `@code_cell`:
```julia
println("Second cell!")

function foo(arguments)
end
```

The Julia parser gives the following tree:
```
(source_file ; [0, 0] - [10, 0]
  (line_comment) ; [0, 0] - [0, 36]
  (line_comment) ; [1, 0] - [1, 9]
  (call_expression ; [3, 0] - [3, 23]
    (identifier) ; [3, 0] - [3, 7]
    (argument_list ; [3, 7] - [3, 23]
      (string_literal))) ; [3, 8] - [3, 22]
  (function_definition ; [5, 0] - [7, 3]
    (signature ; [5, 9] - [5, 23]
      (call_expression ; [5, 9] - [5, 23]
        (identifier) ; [5, 9] - [5, 12]
        (argument_list ; [5, 12] - [5, 23]
          (identifier))))) ; [5, 13] - [5, 22]
  (line_comment)) ; [9, 0] - [9, 63]
```
So I wrote the following query:
```
(
 source_file 
 (
  (_) @code_cell._inner (#not-has-type? @code_cell._inner line_comment)
  ) @code_cell
)
```
However, this captures individually each statement between comments as `@code_cell`, I would like to merge them together and have only one `@code_cell`. I tried using the `+` operator, but with little success. The following query captures individually the three first lines of my code sample:
```
(
 source_file 
 (
  (_) @code_cell._inner (#not-has-type? @code_cell._inner line_comment)
  )+ @code_cell
)
```

I am obviously doing something wrong here, but I do not understand what. How can I write my query correctly?

**Comment by clason** [comment, marked as answer]
That is impossible with queries, I'm afraid. You can only capture nodes, not "stuff between nodes". You'll have to fake it using `metadata`.

**Reply by Klafyvel** [reply]
I see, thanks! Can you explain why the `+` operator does not work in that case?

**Reply by clason** [reply]
No, because that depends on the implementation, not the queries themselves.
